sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/BufferedPublisher.cs

```diff
@@ -0,0 +1,175 @@
+
+                    // Start concurrent enqueuing tasks
+
+                    if (_testConfiguration.ConcurrentSends > 1)
+                    {
+                        for (var index = 0; index < _testConfiguration.ConcurrentSends - 1; ++index)
+                        {
+                            enqueueTasks.Add(Task.Run(async () =>
+                            {
+                                while (!cancellationToken.IsCancellationRequested)
+                                {
+                                    await PerformEnqueue(producer, cancellationToken).ConfigureAwait(false);
+
+                                    if ((_testConfiguration.ProducerPublishingDelay.HasValue) && (_testConfiguration.ProducerPublishingDelay.Value > TimeSpan.Zero))
+                                    {
+                                        await Task.Delay(_testConfiguration.ProducerPublishingDelay.Value, backgroundCancellationSource.Token).ConfigureAwait(false);
+                                    }
+                                }
+                            }));
+                        }
+
+                        // Perform one of the sends in the foreground, which will allow easier detection of a
+                        // processor-level issue.
+
+                        while (!cancellationToken.IsCancellationRequested)
+                        {
+                            try
+                            {
+                                await PerformEnqueue(producer, cancellationToken).ConfigureAwait(false);
+
+                                if ((_testConfiguration.ProducerPublishingDelay.HasValue) && (_testConfiguration.ProducerPublishingDelay.Value > TimeSpan.Zero))
+                                {
+                                    await Task.Delay(_testConfiguration.ProducerPublishingDelay.Value, cancellationToken).ConfigureAwait(false);
+                                }
+                            }
+                            catch (TaskCanceledException)
+                            {
+                                backgroundCancellationSource.Cancel();
+                                await Task.WhenAll(enqueueTasks).ConfigureAwait(false);
+                            }
+                        }
+                    }
+                }
+                catch (TaskCanceledException)
+                {
+                    // No action needed, the cancellation token has been cancelled.
+                }
+                catch (Exception ex) when
+                    (ex is OutOfMemoryException
+                    || ex is StackOverflowException
+                    || ex is ThreadAbortException)
+                {
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    // If this catch is hit, it means the producer has restarted, collect metrics.
+
+                    _metrics.Client.GetMetric(_metrics.ProducerRestarted).TrackValue(1);
```

Comment:
Does this increment the count or are we collecting these as individual events and then aggregating via Kusto?

Reply:
The GetMetric method is cool it aggregates (in this case sums) up all the values sent via "TrackValue" and sends them all at once:
"In summary GetMetric() is the recommended approach since it does pre-aggregation, it accumulates values from all the Track() calls and sends a summary/aggregate once every minute. This can significantly reduce the cost and performance overhead by sending fewer data points, while still collecting all relevant information." [ref](https://docs.microsoft.com/en-us/azure/azure-monitor/app/get-metric)
The long-term aggregation like total sum is done in Kusto.